src/go/rpk/pkg/cli/cmd/debug/bundle/bundle_k8s_linux.go

```diff
@@ -213,18 +213,29 @@ func saveSingleAdminAPICalls(ctx context.Context, ps *stepParams, fs afero.Fs, c
 				continue
 			}
 
+			aName := sanitizeName(a)
 			r := []func() error{
 				func() error {
-					return requestAndSave(ctx, ps, fmt.Sprintf("admin/node_config_%v.json", a), cl.RawNodeConfig)
+					return requestAndSave(ctx, ps, fmt.Sprintf("admin/node_config_%v.json", aName), cl.RawNodeConfig)
 				},
 				func() error {
-					return requestAndSave(ctx, ps, fmt.Sprintf("admin/cluster_view_%v.json", a), cl.ClusterView)
+					return requestAndSave(ctx, ps, fmt.Sprintf("admin/cluster_view_%v.json", aName), cl.ClusterView)
 				},
 				func() error {
-					return requestAndSave(ctx, ps, fmt.Sprintf("metrics/metric_%v.txt", a), cl.PrometheusMetrics)
+					err := requestAndSave(ctx, ps, fmt.Sprintf("metrics/%v/t0_metric.txt", aName), cl.PrometheusMetrics)
+					if err != nil {
+						return err
+					}
+					time.Sleep(metricInterval)
```

Comment:
How long does bundle take today? This basically enforces a hard minimum of 15s by default (10 if dropping). Wdyt about hooking into the command's context cancelation and requesting/saving prometheus metrics immediately once the command context is canceled?

```go
select {
case <-time.After(metricInterval):
case <-cmd.Context().Done():
```
roughly

here and below